packages/react/src/Details/__tests__/Details.test.tsx

```diff
@@ -102,4 +102,65 @@ describe('Details', () => {
 
     expect(getByTestId('summary')).toHaveTextContent('Open')
   })
+
+  it('Adds default summary if no summary supplied', async () => {
+    const {getByText} = render(<Details data-testid="details">content</Details>)
+
+    expect(getByText('See Details')).toBeInTheDocument()
+    expect(getByText('See Details').tagName).toBe('SUMMARY')
+  })
+
+  it('Adds summary element when supplied as prop', async () => {
+    const {getByText} = render(
+      <Details summary="test summary" data-testid="details">
+        content
+      </Details>,
+    )
+
+    expect(getByText('test summary')).toBeInTheDocument()
+    expect(getByText('test summary').tagName).toBe('SUMMARY')
+  })
+
+  it('Ignores summary prop when child exists', async () => {
+    const {queryByText, getByText} = render(
+      <Details summary="test summary" data-testid="details">
+        <Details.Summary>custom summary</Details.Summary>
+        content
+      </Details>,
+    )
+
+    expect(queryByText('test summary')).toBeNull()
+    expect(queryByText('See Details')).toBeNull()
+    expect(getByText('custom summary')).toBeInTheDocument()
+    expect(getByText('custom summary').tagName).toBe('SUMMARY')
+  })
+
+  it('Does not add summary if supplied as different element', async () => {
+    const {getByTestId, queryByText} = render(
+      <Details data-testid="details">
+        <Button as="summary" data-testid="summary">
+          custom summary
+        </Button>
+        content
+      </Details>,
+    )
+
+    expect(queryByText('See Details')).toBeNull()
+    expect(getByTestId('summary')).toBeInTheDocument()
```

Comment:
(just a note) TIL you can't `getByRole('summary')`, it's inferred as a button instead!